core/ecschema-locaters/src/StubSchemaXmlFileLocater.ts

```diff
@@ -84,7 +84,8 @@ export class StubSchemaXmlFileLocater extends SchemaFileLocater implements ISche
       return undefined;
 
     const maxCandidate = candidates.sort(this.compareSchemaKeyByVersion)[candidates.length - 1];
-    const alias = this.getSchemaAlias(maxCandidate.schemaText!);
+    // Note: if maxCandidate.schemaText is undefined, the "" passed to getSchemaAlias will throw an ECSchemaError.
+    const alias = this.getSchemaAlias(maxCandidate.schemaText ?? "");
```

Comment:
This is problematic.  findEligibleSchemaKeys always fills out the schemaText.  Some code uses the schemaText and other code loads it all over again.  It looks like we may load the text from disk more than once and in some cases we may hold onto the text long after we need it (when we construct the schema with a SchemaFileKey)

This should be rationalized and fixed

Reply:
@ColinKerr I'm assuming you are referring to the other locaters (json and xml)?  Looks like we may read it twice. The StubSchemaXmlFileLocater looks ok to me.